pysignalscope/scope.py

```diff
@@ -419,6 +477,13 @@ def power(cls, channel_voltage: 'Scope', channel_current: 'Scope', channel_label
         :return: power in a dataset
         :rtype: Scope
         """
+        if not isinstance(channel_voltage, Scope):
+            raise TypeError("channel_voltage must be type Scope.")
+        if not isinstance(channel_current, Scope):
+            raise TypeError("channel_current must be type Scope.")
+        if not isinstance(channel_label, str):
+            raise TypeError("channel_label must be type str.")
```

Comment:
If the channel_label is None, means not entered, is this considered as string too? Otherwise it makes no sence, that you check later, if the channel_label is none in 488

Reply:
Updated. This label check is the result for the new label after the multiplication (check for None was missing, it is fixed now).

Line 488 checks the input labels of current and voltage. It just generates a default label for the new Scope object in case of no new label is given.